src/openrct2/actions/RideSetStatus.hpp

```diff
@@ -77,7 +77,14 @@ DEFINE_GAME_ACTION(RideSetStatusAction, GAME_COMMAND_SET_RIDE_STATUS, GameAction
             res->ErrorMessage = STR_NONE;
             return res;
         }
-
+        if (_status >= 4 || _status <= -1)
```

Comment:
status is a unsigned so the second check isn't required as it can never be negative.

We try to avoid magic numbers like that 4 but I'm going to guess we haven't defined a ride status count. Could you add RIDE_STATUS_COUNT to https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/ride/Ride.h#L632 and use RIDE_STATUS_COUNT instead of 4.

Reply:
In this case, `_status` gets used as an index for `_StatusErrorTitles`, so instead of using the defines you could use `std::size(_StatusErrorTitles)`.
Of course, this static array needs to be equal in size, so introducing `RIDE_STATUS_COUNT` and adding a `static_assert` there would be useful.

Reply:
Wouldn't a static_assert stop the game from continuing? As the point of this change is to tell the JS operator that they chose an invalid status rather than crashing the game.

Reply:
Would it also be relevant to add to the error message something like:
"Allowable values are range from zero to three"